crates/bevy_utils/src/option_ext.rs

```diff
@@ -0,0 +1,90 @@
+//! Extensions to the [`Option`] type used in Bevy.
+
+use crate::alloc::boxed::Box;
+use core::{error::Error, fmt};
+
+/// A custom type which implements [`Error`], used to indicate that an `Option` was `None`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct NoneError;
+
+impl fmt::Display for NoneError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Unexpected None value.")
+    }
+}
+
+impl Error for NoneError {}
+
+/// Extension trait for [`Option`].
+pub trait OptionExt<T> {
```

Comment:
For clarity I think we can afford to split this into two different traits: `Unpack` and `Assume`.

Reply:
Resolved in [210faa7](https://github.com/bevyengine/bevy/pull/17739/commits/210faa7418348289342a517c183eca8e6f7c30c7).